index.js

```diff
@@ -111,6 +111,28 @@ Babel.prototype.optionsHash = function() {
       hash[key] = (typeof value === 'function') ? (value + '') : value;
     }
 
+    if (options.plugins) {
+      hash.plugins = [];
```

Comment:
Sure that seems like a good addition to what is here, but I don't think it replaces the mechanim/concept in this PR already (since the plugins config needs to be taken into account).